kura/org.eclipse.kura.core.net/src/main/java/org/eclipse/kura/core/net/modem/ModemInterfaceAddressConfigImpl.java

```diff
@@ -39,63 +39,36 @@ public void setNetConfigs(List<NetConfig> configs) {
     }
 
     @Override
-    public boolean equals(Object obj) {
-
-        if (this == obj) {
-            return true;
-        }
-
-        if (!super.equals(obj)) {
-            return false;
-        }
-
-        if (!(obj instanceof ModemInterfaceAddressConfig)) {
-            return false;
-        }
-
-        ModemInterfaceAddressConfig other = (ModemInterfaceAddressConfig) obj;
-
-        if (!compare(getSignalStrength(), other.getSignalStrength())) {
-            return false;
-        }
-
-        if (!compare(isRoaming(), other.isRoaming())) {
-            return false;
-        }
-
-        if (!compare(getConnectionStatus(), other.getConnectionStatus())) {
-            return false;
-        }
-
-        if (getBytesTransmitted() != other.getBytesTransmitted()) {
-            return false;
-        }
-
-        if (getBytesReceived() != other.getBytesReceived()) {
-            return false;
-        }
-
-        if (!compare(getConnectionType(), other.getConnectionType())) {
-            return false;
-        }
-
-        List<NetConfig> thisNetConfigs = getConfigs();
-        List<NetConfig> otherNetConfigs = other.getConfigs();
-
-        if (thisNetConfigs.size() != otherNetConfigs.size()) {
-            return false;
-        }
-        if (!thisNetConfigs.containsAll(otherNetConfigs)) {
-            return false;
-        }
-        if (!otherNetConfigs.containsAll(thisNetConfigs)) {
-            return false;
-        }
-
-        return true;
-    }
-
-    @Override
+	public int hashCode() {
+		final int prime = 31;
+		int result = 1;
+		result = prime * result + ((m_configs == null) ? 0 : m_configs.hashCode());
+		return result;
+	}
+
+	@Override
+	public boolean equals(Object obj) {
+		if (this == obj) {
+			return true;
+		}
+		if (!super.equals(obj)) {
+			return false;
+		}
+		if (!(obj instanceof ModemInterfaceAddressConfigImpl)) {
```

Comment:
As far as I remember this is how Eclipse generates hashCode() and equals(), but I agree that making the "cheaper" check should be done earlier. I'll fix it.